bin/ofs/tests/common/mod.rs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::{
+    collections::HashMap,
+    env,
+    process::{Child, Command},
+};
+
+use assert_cmd::cargo::CommandCargoExt;
+use tempfile::TempDir;
+use test_context::AsyncTestContext;
+
+pub(crate) struct OfsTestContext {
```

Comment:
Maintaining those test cases seems neither easy nor realistic. How about just mount ofs in workflow and calling `ls`, `touch`, `cat`? Or even `fio`? I believe this is the way how users uses `ofs`.

Reply:
Honestly I don't assume that user uses posix CLI tools to interact with `ofs`, and there may be some cases that aren't covered using these CLI tools to test

Reply:
> Honestly I don't assume that user uses posix CLI tools to interact with `ofs`

Can you explain a bit? AFAIK, Users can't interact with `ofs` directly. They must mount `ofs` locally and use tools to interact the fs like `ls`, `cat`.

Reply:
User can use tools such as `ls` and `cat`, but also can use some GUI file manager like `dolphin`.

Reply:
> User can use tools such as `ls` and `cat`, but also can use some GUI file manager like `dolphin`.

You're correct. But if an fs works correctly on `ls`, `cat` or `pjdfstest`， all GUI file manager should work too.

However, it's unlikely that users will use `ofs` by writing Rust code (as we do in the tests). I'm not in favor of writing ofs tests in Rust; they're difficult to maintain and don't accurately reflect real-world use cases.

Reply:
I think we can use `pjdfstest` or `LTP` for tests (by ignoring not supported tests) instead of implement tests cases by hand.

Reply:
> I think we can use `pjdfstest` or `LTP` for tests (by ignoring not supported tests) instead of implement tests cases by hand.

Yeah， I think  POSIX compatibility is our `ofs` goal, we need to `pjdfstest` tool help us to verify all of this.

>  User can use tools such as ls and cat, but also can use some GUI file manager like dolphin.

For the `ofs` tools， `fusemount3` and mount DIR are input interfaces, So I think our test needs to cover all of these interfaces.

Reply:
> I think POSIX compatibility is our `ofs` goal, we need to `pjdfstest` tool help us to verify all of this.

`POSIX compatibility` is not our goal. We just use pjdfstest for testing.

Reply:
I tried migrating Rust test cases to `pjdfstest`, but `pjdfstest` is too strict that `ofs` and `OpenDAL` lack much ability to pass the test.

Reply:
You are right. [mountpoint-s3](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/tests/fuse_tests/write_test.rs) also writes their own test suites.